### PR TITLE
[SDK] Add pagination to getOwned for ERC721 & 1155

### PR DESCRIPTION
**Pull request by kien-ngo**
## Problem solved

Add optional pagination for `getOwned`.

## Changes made

- [x] Public API changes: list the public API changes made if any
- [ ] Internal API changes: explain the internal logic changes

## How to test

- [x] Automated tests: link to unit test file
- [ ] Manual tests: step by step instructions on how to test

### CODE REVIEW

File: packages/sdk/src/evm/core/classes/erc-721-enumerable.ts [hunk 1]
```diff
@@ -46,10 +50,20 @@ export class Erc721Enumerable implements DetectableFeature {
...
    */
-  public async all(walletAddress?: AddressOrEns): Promise<NFT[]> {
-    const tokenIds = await this.tokenIds(walletAddress);
+  public async all(
+    walletAddress?: AddressOrEns,
+    queryParams?: QueryAllParams,
+  ): Promise<NFT[]> {
+    let tokenIds = await this.tokenIds(walletAddress);
+    if (queryParams) {
+      const page = queryParams?.start || 0;
```
Comment (joaquim-verges) [inline packages/sdk/src/evm/core/classes/erc-721-enumerable.ts:62]: keep the var names consistent, `page` -> `start`

File: packages/sdk/src/evm/core/classes/erc-721-enumerable.ts [hunk 2]
```diff
@@ -46,10 +50,20 @@ export class Erc721Enumerable implements DetectableFeature {
...
+      const count = queryParams?.count || DEFAULT_QUERY_ALL_COUNT;
+      const startIndex = page * count;
```
Comment (joaquim-verges) [inline packages/sdk/src/evm/core/classes/erc-721-enumerable.ts:64]: this doesnt seem right?

startIndex should be `start` - make sure this is covered by unit tests

File: packages/sdk/src/evm/core/classes/erc-721.ts [hunk 3]
```diff
@@ -415,25 +418,36 @@ export class Erc721<
...
         walletAddress || this.contractWrapper.getSignerAddress(),
         this.getAllOwners(),
       ]);
+      let ownedTokens = (allOwners || []).filter(
+        (i) => address?.toLowerCase() === i.owner?.toLowerCase(),
+      );
+      if (queryParams) {
+        const page = queryParams?.start || 0;
+        const count = queryParams?.count || DEFAULT_QUERY_ALL_COUNT;
+        const startIndex = page * count;
```
Comment (joaquim-verges) [inline packages/sdk/src/evm/core/classes/erc-721.ts:446]: same issue here

File: packages/sdk/src/evm/core/classes/erc-721a-queryable.ts [hunk 4]
```diff
@@ -50,8 +54,17 @@ export class Erc721AQueryable implements DetectableFeature {
...
-    const tokenIds = await this.tokenIds(walletAddress);
+  public async all(
+    walletAddress?: AddressOrEns,
+    queryParams?: QueryAllParams,
+  ): Promise<NFT[]> {
+    let tokenIds = await this.tokenIds(walletAddress);
+    if (queryParams) {
+      const page = queryParams?.start || 0;
+      const count = queryParams?.count || DEFAULT_QUERY_ALL_COUNT;
+      const startIndex = page * count;
```
Comment (joaquim-verges) [inline packages/sdk/src/evm/core/classes/erc-721a-queryable.ts:65]: and here

**Comment by kien-ngo** [comment]
@joaquim-verges Thanks again for taking your time to review this. This code of mine:
```
const start = queryParams?.start || 0;
const count = queryParams?.count || DEFAULT_QUERY_ALL_COUNT;
const startIndex = start * count;
tokenIds = tokenIds.slice(startIndex, startIndex + count);
```
is based on how I think the pagination for `getOwned` should work as demonstrated in the image:
image

which is a bit different to how it works in `getAll`. Do you think there's anything wrong with this approach?

p/s: I have set up the unit tests here: 
ERC1155: https://github.com/thirdweb-dev/js/blob/c7476e885ad63c6ebdb53011bb1d70ff134f85ee/packages/sdk/test/evm/edition.test.ts#L293-L333

ERC721: https://github.com/thirdweb-dev/js/blob/c7476e885ad63c6ebdb53011bb1d70ff134f85ee/packages/sdk/test/evm/nft.test.ts#L291-L319

**Comment by joaquim-verges** [comment]
@kien-ngo you're still considering `start` as the page number instead of the actual start tokenId.

This is a different behavior from the existing `erc721.getAll()`, and will the dashboard pagination and any customer that uses pagination today to implement 2 different kinds of paginations.

please keep the current behavior and keep it consistent between getAll and getOwned.

**Comment by kien-ngo** [comment]
Ah I see. Okay so the `start` should be one of the "owned" tokenIds to make it consistent with getAll. Will update my code today ^

**Comment by joaquim-verges** [comment]
@kien-ngo to be clear - `start` doesn't need to be the actual `tokenId` number, but rather the *index* of that token in the array of owned tokenIds.

for getAll it just happens to be that index == tokenId, but not the case for getOwned.

In general the way to think about it is:

let's say i owned 20 NFTs total, calling

`getOwned("0x...", { start: 8, count: 3 })`

should give me my 8th, 9th and 10th owned NFT (regardless of their tokenID)

lmk if that makes sense

**Comment by kien-ngo** [comment]
Yes ser it does. Thanks again. I will update the code and the tests

**Comment by kien-ngo** [comment]
@joaquim-verges Hi there, I updated the code & added unit tests. Please have a look when you have time thank you.

**Review by joaquim-verges — APPROVED** [review]
Looks good! Thanks @kien-ngo

**Pull request merged**